__tests__/workers/newView.ts

```diff
@@ -159,14 +159,24 @@ describe('reading streaks', () => {
   ) => {
     await prepareTest(currentDate, previousDate, previousStreak);
 
-    const streak = await con.getRepository(UserStreak).findOne({
-      where: {
-        userId: 'u1',
-        lastViewAt: new Date(currentDate),
-      },
+    const user = await con.getRepository(User).findOne({
+      where: { id: 'u1' },
+      relations: ['streak'],
+    });
+
+    const streak = await user.streak;
+
+    const lastView = await con.getRepository(View).findOne({
+      where: { userId: 'u1' },
+      order: { timestamp: 'DESC' },
     });
+
     for (const key in expectedStreak) {
-      expect(streak?.[key]).toEqual(expectedStreak[key]);
+      if (key === 'lastViewAt') {
+        expect(streak?.[key]).toEqual(lastView.timestamp);
+      } else {
+        expect(streak?.[key]).toEqual(expectedStreak[key]);
+      }
```

Comment:
This change does not seem correct to me. Previously we were comparing the lastViewAt on the `user_streak` table, now we do it against the `view` table. 

If we want to keep the same test behavior, we should take the `lastViewAt` from the `streak` object you create above.

Reply:
we actually don't want to keep the same behaviour, the goal here is not to compare against the date coming in, but against the timestamp of the last view record. 

essentially what we are doing here is asserting that after the worker runs, our streak `lastViewAt` correctly matches the  `timestamp` of the last view -- we intentionally don't want to compare it against the date being passed in